src/Avalonia.Input/PenDevice.cs

```diff
@@ -0,0 +1,415 @@
+using System;
+using System.Linq;
+using System.Reactive.Linq;
+using Avalonia.Input.Raw;
+using Avalonia.Interactivity;
+using Avalonia.Platform;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Input
+{
+    /// <summary>
+    /// Represents a pen/stylus device.
+    /// </summary>
+    public class PenDevice : IPenDevice, IDisposable
+    {
+        private int _clickCount;
+        private Rect _lastClickRect;
+        private ulong _lastClickTime;
+
+        private readonly Pointer _pointer;
+        private bool _disposed;
+        private PixelPoint? _position; 
+
+        public PenDevice(Pointer? pointer = null)
```

Comment:
Each new contact should be represented by a new pointer. The previous Pointer is no longer valid when pen device loses contact with the tabled.